File: app/src/main/java/com/jay/currencyconverter/BaseApplication.java
package com.jay.currencyconverter;

import android.app.Application;

import com.jay.currencyconverter.di.BaseComponent;
import com.jay.currencyconverter.di.BaseModule;
import com.jay.currencyconverter.di.DaggerBaseComponent;
import com.jay.currencyconverter.di.DaggerNetworkComponent;
import com.jay.currencyconverter.di.NetworkComponent;

public class BaseApplication extends Application {

    private static BaseComponent baseComponent;
    private static NetworkComponent networkComponent;

    @Override
    public void onCreate() {
        baseComponent = DaggerBaseComponent.builder()
                .baseModule(new BaseModule(this))
                .build();

        networkComponent = DaggerNetworkComponent.create();
        super.onCreate();
    }

    public static BaseComponent getBaseComponent() {
        return baseComponent;
    }

    public static NetworkComponent getNetworkComponent() {
        return networkComponent;
    }
}

File: app/src/main/java/com/jay/currencyconverter/BaseApplication.kt
package com.jay.currencyconverter

import android.app.Application
import androidx.room.Room
import com.jay.currencyconverter.di.base.BaseComponent
import com.jay.currencyconverter.di.base.BaseModule
import com.jay.currencyconverter.di.base.DaggerBaseComponent
import com.jay.currencyconverter.di.network.DaggerNetworkComponent
import com.jay.currencyconverter.di.network.NetworkComponent
import com.jay.currencyconverter.repository.room.DataBase

class BaseApplication : Application() {

    override fun onCreate() {
        baseComponent = DaggerBaseComponent.builder().baseModule(BaseModule(application = this)).build()
        networkComponent = DaggerNetworkComponent.create()
        dataBase = Room.databaseBuilder(this, DataBase::class.java, "DataBase").build()
        super.onCreate()
    }

    companion object {
        lateinit var baseComponent: BaseComponent
        lateinit var networkComponent: NetworkComponent
        lateinit var dataBase: DataBase
    }
}